controllers/vspheremachine_controller.go

```diff
@@ -388,7 +389,11 @@ func (r machineReconciler) reconcileNormalPre7(ctx *context.MachineContext) (run
 			}))
 
 		// Instruct the VSphereVM to use the CAPI bootstrap data resource.
-		vm.Spec.BootstrapRef = ctx.Machine.Spec.Bootstrap.ConfigRef
+		// TODO: BootstrapRef field should be replaced with BootstrapSecret of type string
+		vm.Spec.BootstrapRef = &corev1.ObjectReference{
+			Name:      *ctx.Machine.Spec.Bootstrap.DataSecretName,
```

Comment:
what if the `DataSecretName` is nil ? the VM shouldn't have a `BootstrapRef` if the secret isn't available yet

Reply:
@yastij Yes, I thought about it and wanted to add a nil check. But I realized that my changes are in `reconcileNormalPre7` function and and there is already a check for nil `DataSecretName` just befoe this function is called:
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/controllers/vspheremachine_controller.go#L312-L318

Reply:
oh that's all good then. 

I know it'll work like this as is, but can you add for completeness:
```
APIVersion: "v1",
Kind:       "Secret",
```

Reply:
@yastij  I'm not sure what you are asking here. Do you want me to change the type for `BootstrapRef`?

Reply:
@dhawal55 - we need the `APIVersion` and `Kind` fields of the `ObjectReference` set

Reply:
@yastij Thank you. I updated the PR.